cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -1185,24 +1183,36 @@ public synchronized void userTriggeredStopExecution() {
     return result;
   }
 
-  private static Set<String> topicsForReplicationFactorChange(Pattern topicPattern, Cluster cluster, short replicationFactor) {
-    Set<String> topics = cluster.topics().stream().filter(t -> topicPattern.matcher(t).matches()).collect(Collectors.toSet());
-    // Ensure there are topics matching the requested topic pattern.
-    if (topics.isEmpty()) {
-      throw new IllegalStateException("There is no topic in cluster matching pattern " + topicPattern);
+  private static Map<Short, Set<String>> topicsForReplicationFactorChange(Map<Short, Set<Pattern>> topicPatternsByReplicationFactor,
+                                                                          Cluster cluster) {
+    Map<Short, Set<String>> topicsToChangeByReplicationFactor = new HashMap<>(topicPatternsByReplicationFactor.size());
+    for (Map.Entry<Short, Set<Pattern>> entry : topicPatternsByReplicationFactor.entrySet()) {
+      short replicationFactor = entry.getKey();
+      for (Pattern topicPattern : entry.getValue()) {
+        Set<String> topics = cluster.topics().stream().filter(t -> topicPattern.matcher(t).matches()).collect(Collectors.toSet());
+        // Ensure there are topics matching the requested topic pattern.
+        if (topics.isEmpty()) {
+          throw new IllegalStateException("There is no topic in cluster matching pattern " + topicPattern);
+        }
+        Set<String> topicsToChange = topics.stream()
```

Comment:
Not sure where is the best place in the method for this suggestion, but something should validate whether there is an intersection between any of the topic sets built by the different provided patterns.

Otherwise there will be an insidious bug where two patterns are provided that both match `topicA`, and it is not deterministic what replication factor `topicA` will get.

Reply:
The current logic is that if request URL contains topic and replication factor, whatever specified in request body will be ignored. 

The idea is that the two approach to specify request should be mutual-exclusive. I document this in [Javadoc](https://github.com/linkedin/cruise-control/blob/b818bc16eed52544d7b584e5bf7caa6ef3619ad2/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/TopicConfigurationParameters.java#L30), once this patch get merged, I will update wiki page about it.

Reply:
Ah, I might be mis-parsing this method then.  
Does this PR allow for the body to specify more than one replication factor : pattern mapping, or only one?

If only one, then my comment is a non-issue.  
If more than one, then my concern is that there may be a topic set intersection, such that it's ambiguous which replication factor a colliding topic should have.

Reply:
Ah, I see. good point! I will add a sanity check that there will be no more  than one target replication factor for each topic.